### PR TITLE
Fix 4363 modify kanban menu

### PR DESCRIPTION
**Pull request by kiridarivaki**
**Changes:**

- Changed -/+ to eye and eye off icons
- Changed menu width to 200px
- Created separate menu for hidden fields
- Added Edit Fields option to hidden fields menu
- Added test file MenuItemSelectTag (wasn't included in the issue)

As this is my first pr, feedback is very welcome!
**Note:** 
These changes cover most of #4363 . I left out the implementation of the RightIcon in the "Hidden Fields" menu item.

### CODE REVIEW

**Comment by FelixMalfait** [comment]
@kiridarivaki This is great work for a first PR :) ! Thank you!

Could you please check the view switcher? I think it had a negative impact there

The rest looks good

Screenshot 2024-05-14 at 09 50 39

**Comment by kiridarivaki** [comment]
@FelixMalfait Thank you for noticing! Will work on it now:)

**Comment by FelixMalfait** [comment]
@kiridarivaki can you try adding a view?

**Comment by kiridarivaki** [comment]
@FelixMalfait oh sorry you meant the misplacement of the pencil icon is the issue? Now i see it, thanks!

**Comment by FelixMalfait** [comment]
@kiridarivaki yes exactly :)

File: packages/twenty-front/src/modules/object-record/record-index/options/components/RecordIndexOptionsDropdownContent.tsx [hunk 1]
```diff
@@ -158,6 +158,7 @@ export const RecordIndexOptionsDropdownContent = ({
             onClick={() => handleSelectMenu('hiddenFields')}
             LeftIcon={IconEyeOff}
             RightIcon={IconChevronRight}
+            hasRightIcon={true}
```
Comment (FelixMalfait) [inline packages/twenty-front/src/modules/object-record/record-index/options/components/RecordIndexOptionsDropdownContent.tsx:161]: why do you introduce this and not look at RightIcon directly?

File: packages/twenty-front/src/modules/ui/navigation/menu-item/components/MenuItem.tsx [hunk 2]
```diff
@@ -63,14 +65,18 @@ export const MenuItem = ({
...
       </StyledMenuItemLeftContent>
-      <div className="hoverable-buttons">
-        {showIconButtons && (
-          <LightIconButtonGroup iconButtons={iconButtons} size="small" />
-        )}
-      </div>
       <StyledMenuItemRightContent>
-        <MenuItemLeftContent LeftIcon={RightIcon ?? undefined} text={''} />
+        {hasRightIcon && (
+          <MenuItemLeftContent LeftIcon={RightIcon ?? undefined} />
```
Comment (FelixMalfait) [inline packages/twenty-front/src/modules/ui/navigation/menu-item/components/MenuItem.tsx:70]: LeftIcon={RightIcon ?

File: packages/twenty-front/src/modules/ui/navigation/menu-item/internals/components/MenuItemLeftContent.tsx [hunk 3]
```diff
@@ -14,7 +14,7 @@ type MenuItemLeftContentProps = {
   className?: string;
   LeftIcon: IconComponent | null | undefined;
   showGrip?: boolean;
-  text: string;
+  text?: string;
```
Comment (FelixMalfait) [inline packages/twenty-front/src/modules/ui/navigation/menu-item/internals/components/MenuItemLeftContent.tsx:17]: What does it mean to have a MenuItem without content?

File: packages/twenty-front/src/modules/ui/navigation/menu-item/components/MenuItem.tsx (hunk 2 above)
Comment (kiridarivaki) [reply packages/twenty-front/src/modules/ui/navigation/menu-item/components/MenuItem.tsx:70]: Oh yes, I wanted to ask you about it, as I saw that there is a MenuItemLeftContent but not a MenuItemRightContent. Should I create MenuItemRightContent? I though that adding a RightIcon prop to the MenuItemLeftContent isn't a good practice either. What is preferable? I'm sorry for the mediocre code quality, trying my best as I'm still learning :)

File: packages/twenty-front/src/modules/ui/navigation/menu-item/internals/components/MenuItemLeftContent.tsx (hunk 3 above)
Comment (kiridarivaki) [reply packages/twenty-front/src/modules/ui/navigation/menu-item/internals/components/MenuItemLeftContent.tsx:17]: Should i make the text equal to an empty string? I thought text was not needed when creating the right content of the menu item as it's only an icon (line 70 in MenuItem.tsx)

Comment (charlesBochet) [reply packages/twenty-front/src/modules/ui/navigation/menu-item/internals/components/MenuItemLeftContent.tsx:17]: see MenuItemSelect.tsx to see how to add a Right icon and implement it in MenuItem direclty. Do not MenuItemLeftContent

File: packages/twenty-front/src/modules/object-record/record-index/options/components/RecordIndexOptionsDropdownContent.tsx (hunk 1 above)
Comment (FelixMalfait) [reply packages/twenty-front/src/modules/object-record/record-index/options/components/RecordIndexOptionsDropdownContent.tsx:161]: Thanks! what I ment was `hasRightIcon` seems synonymous of `!isUndefinedOrNull(RightIcon)` no? maybe I'm missing something. But it doesn't seem like a clean component api if you can have rightIcon=XXX and hasRightIcon=false (contradiction)

**Comment by FelixMalfait** [comment]
Thanks for the updates! I see there's a small conflict with main, could you also please merge the most recent version of main? Hopefully we can merge that soon after the small fixes :)

**Comment by kiridarivaki** [comment]
@FelixMalfait @charlesBochet Thank you for the feedback! I'll try my best to do these fixes and improve :)

File: packages/twenty-front/src/modules/ui/navigation/menu-item/components/MenuItem.tsx (hunk 2 above)
Comment (kiridarivaki) [reply packages/twenty-front/src/modules/ui/navigation/menu-item/components/MenuItem.tsx:70]: @charlesBochet check Discord :)

**Review by FelixMalfait — APPROVED** [review]
Thank you!

**Pull request merged**